src/test/java/com/puppycrawl/tools/checkstyle/DefaultLoggerTest.java

```diff
@@ -81,6 +85,33 @@ public void testCtorWithTwoParameters() {
                 .contains("java.lang.IllegalStateException: upsss");
     }
 
+    /**
+     * We keep this test for 100% coverage. Until #12873.
+     */
+    @Test
+    public void testCtorWithTwoParametersCloseStreamOptions() {
+        final OutputStream infoStream = new ByteArrayOutputStream();
+        final DefaultLogger dl = new DefaultLogger(infoStream,
+                AutomaticBean.OutputStreamOptions.CLOSE);
+        final boolean closeInfo = TestUtil.getInternalState(dl, "closeInfo");
+
+        assertWithMessage("closeInfo should be true")
+                .that(closeInfo)
+                .isTrue();
+    }
+
+    @Test
+    public void testCtorWithTwoParametersNoneStreamOptions() {
```

Comment:
put brief reason on why we need pure unit test implementation.

Reply:
Done.